ion/marshal.go

```diff
@@ -352,6 +359,31 @@ func (m *Encoder) encodeDecimal(v reflect.Value) error {
 	return m.w.WriteDecimal(d)
 }
 
+func (m *Encoder) encodeWithAnnotation(v reflect.Value) error {
+	original := v
+	fields := fieldsFor(v.Type())
```

Comment:
Might be worth passing these in instead of re-deriving them here?

Reply:
Done.